ruby/ql/lib/codeql/ruby/regexp/ReDoSUtil.qll

```diff
@@ -595,6 +595,49 @@ private class EdgeLabel extends TInputSymbol {
   }
 }
 
+/**
+ * A RegExp term that acts like a plus.
+ * Either it's a RegExpPlus, or it is a range {1,X} where X is >= 30.
+ * 30 has been chosen as a threshold because for exponential blowup 2^30 is enough to get a decent DOS attack.
+ */
+private class EffectivelyPlus extends RegExpTerm {
+  EffectivelyPlus() {
+    this instanceof RegExpPlus
+    or
+    exists(RegExpRange range | range.getLowerBound() = 1 and range.getUpperBound() >= 30 |
```

Comment:
Can you extend this to handle `not exists(range.getUpperBound())` as well? Then we'd catch `{1,}` (1 or more repetitions). Here's [an example in the wild](https://github.com/github/atom.io/blob/66529d5330290cb1fe153e6b5aa6e3b87a3faa3b/app/helpers/sessions_helper.rb#L17).